src/plugins/api-proxy/util/send.js

```diff
@@ -0,0 +1,325 @@
+	const cookie = Object.keys(cookies)
+		.map(name => `${name}=${cookies[name]}`)
+		.join('; ');
+
+	return {
+		cookie,
+		'csrf-token': csrf,
+	};
+}
+
+export function getLanguageHeader(request) {
+	const requestLang = request.getLanguage();
+	const headerLang = request.headers['accept-language'];
+	const acceptLang =
+		requestLang && headerLang
+			? `${requestLang},${headerLang}`
+			: requestLang || headerLang;
+	return acceptLang;
+}
+
+export function parseRequestHeaders(request) {
+	const externalRequestHeaders = {
+		...request.headers,
+		...getAuthHeaders(request),
+		'accept-language': getLanguageHeader(request),
+	};
+
+	delete externalRequestHeaders['host']; // let app server set 'host'
+	delete externalRequestHeaders['accept-encoding']; // let app server set 'accept'
+	delete externalRequestHeaders['content-length']; // original request content-length is irrelevant
+	delete externalRequestHeaders['content-type']; // the content type will be set in buildRequestArgs
+
+	// cloudflare headers we don't want to pass on
+	delete externalRequestHeaders['cf-ray'];
+	delete externalRequestHeaders['cf-ipcountry'];
+	delete externalRequestHeaders['cf-visitor'];
+	delete externalRequestHeaders['cf-connecting-ip'];
+
+	return externalRequestHeaders;
+}
+
+export function parseFormData(payload, uploadsList) {
+	Object.keys(payload).reduce((formData, key) => {
+		const value = payload[key];
+		if (value.filename) {
+			formData[key] = {
+				value: fs.createReadStream(value.path),
+				options: {
+					filename: value.filename,
+					contentType: value.headers['content-type'],
+				},
+			};
+			uploadsList.push(value.path);
+		} else {
+			formData[key] = value;
+		}
+		return formData;
+	}, {});
+}
+
+/*
+ * Translate the incoming Hapi request into an 'opts' object that can be used
+ * to call `request` for the REST API. This function essentially translates the
+ * app server request properties (headers, payload) into corresponding REST API
+ * request properties. The resulting value will be used as the _base_ set of
+ * options for _every_ parallel REST API request made by the platform
+ * corresponding to single incoming request.
+ * 
+ * @return {Object} externalRequestOpts
+ */
+export function getExternalRequestOpts(request) {
+	const baseUrl = request.server.settings.app.api.root_url;
+	const externalRequestOpts = {
+		baseUrl,
+		method: request.method,
+		headers: parseRequestHeaders(request), // make a copy to be immutable
+		mode: 'no-cors',
+		time: true, // time the request for logging
+		agentOptions: {
+			rejectUnauthorized: baseUrl.indexOf('.dev') === -1,
+		},
+	};
+	if (request.mime === 'multipart/form-data') {
+		// the parsed payload includes string key-value pairs for regular inputs,
+		// and file descriptors for file upload inputs { filename, path, headers }
+		// The file descriptors can be converted to readable streams for the API
+		// request.
+		externalRequestOpts.formData = parseFormData(
```

Comment:
this is a change - breaking out the file-transforming logic into its own pure function